src/allmydata/test/test_auth.py

```diff
@@ -138,3 +142,39 @@ def authenticated(avatarId):
                 (avatarId.username, avatarId.rootcap))
         avatarId.addCallback(authenticated)
         return avatarId
+
+
+class AccountURLCheckerTests(unittest.TestCase):
+    valid_password_characters = string.ascii_letters + string.digits + string.punctuation
+    def test_build_multipart(self):
+        header, body = auth.AccountURLChecker._build_multipart(
+            action="authenticate",
+            email="schmoe@joe.org",
+            password=self.valid_password_characters,
+        )
+        ct = InsensitiveDict(header)['content-type']
+        assert ct.startswith('multipart/form-data; boundary=')
+        _, _, boundary = ct.partition('boundary=')
+        assert boundary in body
+        last = '--' + boundary + '--'
+        assert last in body
+        parts = body.split('--' + boundary)
+        assert not any(part.startswith('--') for part in parts[:-1])
+        assert parts[-1].startswith('--')
+        del parts[-1]
+        del parts[0]
+        assert len(parts) == 3
+        assert all('\r\n\r\n' in part.strip() for part in parts)
+
+    def test_get_page(self):
+        checker = auth.AccountURLChecker(None, 'https://httpbin.org/post')
```

Comment:
Okay. I estimate that may be several hours to get right, given that it involves standing up a service, doing port binding, and using Twisted Web.

Reply:
There's a (fairly basic) example of what I mean w/ "in-memory" at src/allmydata/test/web/test_private.py.  Instead of running a TCP server, it just hooks Twisted clients and servers up to each other with in-memory buffers.  This conveniently avoids the part where you need to set up a factory, make robust port binding code, etc.

Reply:
We've decided not to fix this and instead just eliminate the functionality.